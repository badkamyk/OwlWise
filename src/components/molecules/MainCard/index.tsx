import MainHeading from '../../atoms/HeadingMain'
import MainCardContent from '../MainCardContent'
import { MainCardType } from '../MainCardContent/MainCardType'

export default function MainCard({ title, description, src }: MainCardType) {
    return (
        <div className="flex items-center">
            <div className="group relative mx-auto w-96 overflow-hidden rounded-[16px] bg-gray-300 p-[1px] transition-all duration-300 ease-in-out hover:bg-gradient-to-r hover:from-indigo-500 hover:via-purple-500 hover:to-pink-500">
                <div className="group-hover:animate-spin-slow invisible absolute -top-40 -bottom-40 left-10 right-10 bg-gradient-to-r from-transparent via-white/90 to-transparent group-hover:visible"></div>
                <MainCardContent title={title} description={description} src={src} />
            </div>
        </div>
    )
}
