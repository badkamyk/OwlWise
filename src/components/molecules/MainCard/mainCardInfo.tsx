export const mainCardInfo = [
	{
		title: "Analyze",
		description:
			"We analyze your technical and business needs to create a custom solution that fits your needs.",
		src: "https://nuxt.com/assets/home/ux-fast-light.svg",
	},
	{
		title: "Prototype",
		description:
			"We create a prototype of your solution to ensure that it meets your needs and expectations.",
		src: "https://nuxt.com/assets/home/ux-fast-light.svg",
	},
	{
		title: "Develop",
		description:
			"We develop your solution using the latest technologies and best practices to ensure a high-quality product.",
		src: "https://nuxt.com/assets/home/ux-fast-light.svg",
	},
	{
		title: "Test",
		description:
			"We test your solution to ensure that it meets your needs and expectations.",
		src: "https://nuxt.com/assets/home/ux-fast-light.svg",
	},
];
