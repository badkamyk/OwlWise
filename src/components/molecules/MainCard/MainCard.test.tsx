import { render, screen } from "@testing-library/react";
import "intersection-observer";
import { expect, it } from "vitest";
import MainCard from "./index";

it("renders MainCard with props correctly", () => {
	const { container } = render(
		<MainCard
			title={"title"}
			description={"description"}
			src={"https://via.placeholder.com/150"}
		/>
	);
	expect(container.getElementsByClassName("flex items-center")).toBeTruthy();
	expect(
		container.getElementsByClassName(
			"group relative mx-auto w-96 overflow-hidden rounded-[16px] bg-gray-300 p-[1px] transition-all duration-300 ease-in-out hover:bg-gradient-to-r hover:from-indigo-500 hover:via-purple-500 hover:to-pink-500"
		)
	).toBeTruthy();
	expect(
		container.getElementsByClassName(
			"invisible absolute -top-40 -bottom-40 left-10 right-10 bg-gradient-to-r from-transparent via-white/90 to-transparent group-hover:visible group-hover:animate-spin-slow"
		)
	).toBeTruthy();
	expect(screen.getByText("title")).toBeTruthy();
	expect(screen.getByText("description")).toBeTruthy();
	expect(screen.getByRole("img")).toBeTruthy();
	expect(screen.getByRole("img")).toHaveProperty(
		"src",
		"https://via.placeholder.com/150"
	);
	expect(container).toMatchSnapshot();
});
