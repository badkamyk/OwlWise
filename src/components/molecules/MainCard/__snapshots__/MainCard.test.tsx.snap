// Vitest Snapshot v1

exports[`MainCard 1`] = `
<div>
  <div
    class="flex items-center"
  >
    <div
      class="group relative mx-auto w-96 overflow-hidden rounded-[16px] bg-gray-300 p-[1px] transition-all duration-300 ease-in-out hover:bg-gradient-to-r hover:from-indigo-500 hover:via-purple-500 hover:to-pink-500"
    >
      <div
        class="invisible absolute -top-40 -bottom-40 left-10 right-10 bg-gradient-to-r from-transparent via-white/90 to-transparent group-hover:visible group-hover:animate-spin-slow"
      />
      <div
        class="relative rounded-[15px] bg-white p-6"
        style="opacity: 0; transform: translateY(50px);"
      >
        <div
          class="space-y-4"
        >
          <img
            alt="Image with particles"
            src="https://via.placeholder.com/150"
          />
          <p
            class="text-lg font-semibold text-slate-800"
          >
            title
          </p>
          <p
            class="mb-4 text-gray-900 text-md"
          >
            description
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
